src/Xenial.Tasty/Plugins/PluginLoader.cs

```diff
@@ -19,20 +19,34 @@ public Task LoadPlugins(TastyScope scope)
 
             foreach (var pluginAttribute in pluginAttributes)
             {
+#if DEBUG
                 Console.WriteLine($"Loading Plugin: {pluginAttribute.TastyPluginType} from {pluginAttribute.TastyPluginType.Assembly.Location}");
-
-                var @delegate = (Action<TastyScope>)Delegate.CreateDelegate(
-                    typeof(Action<TastyScope>),
-                    null,
-                    pluginAttribute.TastyPluginType.GetMethod(pluginAttribute.TastyPluginEntryPoint, BindingFlags.Static | BindingFlags.Public)
-                );
-                @delegate.Invoke(scope);
-
+#endif
+                try
+                {
+                    var @delegate = (TastyPlugin)Delegate.CreateDelegate(
+                        typeof(TastyPlugin),
+                        null,
+                        pluginAttribute.TastyPluginType.GetMethod(pluginAttribute.TastyPluginEntryPoint, BindingFlags.Static | BindingFlags.Public)
+                    );
+
+                    @delegate.Invoke(scope);
+                }
+                catch (ArgumentException ex)
+                {
+                    throw new InvalidPluginException(@$"Unable to load plugin {pluginAttribute.TastyPluginType} from {pluginAttribute.TastyPluginType.Assembly.Location}.
+The plugin needs to be compatible with delegate {nameof(TastyPlugin)}.
+It must be a static extention method that accepts an {nameof(TastyScope)} and must return an {nameof(TastyScope)}",
```

Comment:
Typos:
```
It must be a static extension method that accepts a {nameof(TastyScope)} and must return a {nameof(TastyScope)}",
```

Reply:
Any planned Tasty localization? Should message templates be in resx?

Reply:
I am not sure if it's worth the effort to focus on localization, cause it's a dev tool in the first place and until now there was nobody asking for it. @hazard999 what are your thoughts on this?

Reply:
English first
I personally don't like localized dev tools, it makes searching for errors and messages a PITA
On the other hand in some areas people lack the language skills or prefer their native language
I won't make it a top prio atm